flycheck.el

```diff
@@ -10516,6 +10517,69 @@ See URL `http://www.gnu.org/software/texinfo/'."
           line-end))
   :modes texinfo-mode)
 
+(flycheck-def-config-file-var flycheck-textlint-config
+    textlint "textlintrc.json"
+  :safe #'stringp)
+
+(flycheck-def-option-var flycheck-textlint-plugin-alist
+    '((markdown-mode . "@textlint/markdown")
+      (gfm-mode . "@textlint/markdown")
+      (t . "@textlint/text"))
+    textlint
+  "An alist mapping major modes to textlint plugins.
+
+Each item is a cons cell `(MAJOR-MODE . PLUGIN)' where MAJOR-MODE
+is a mode `flycheck-textlint' supports and PLUGIN is a textlint plugin.
+
+This needs to be set because textlint plugins are installed seperately,
+and there is no way to check their installation status -- textlint simply
+prints a backtrace."
+  :type '(repeat (choice (cons symbol string)
+                         (cons (const t) string))))
+
+(flycheck-define-checker textlint
+  "A text prose linter using textlint.
+
+See URL `https://textlint.github.io/'."
+  :command ("textlint"
+            (config-file "--config" flycheck-textlint-config)
+            "--format" "unix"
+            ;; get the first matching plugin from plugin-alist
+            "--plugin"
+            (eval (car
+                   (-remove
+                    #'booleanp
+                    (mapcar
+                     (lambda (item)
+                       (let ((mode (car item))
+                             (plugin (cdr item)))
+                         (if (or (and mode (booleanp mode)) ; mode is t
+                                 (derived-mode-p mode))
+                             plugin
+                           nil)))
+                     flycheck-textlint-plugin-alist))))
```

Comment:
What about
```emacs-lisp
(cdr
 (-first
   (pcase-lambda (`(,mode . ,plugin))
     (or (and (booleanp mode) mode) ; mode is t
         (derived-mode-p mode)))
   flycheck-textlint-plugin-alist))
```
instead?

Reply:
I didn't know about `-first`, thanks!

I get a warning that `pcase-lambda` depends on Emacs 25, so I've implemented it with a plain lambda / let.

Reply:
Oh, good call.